File: go/stack.go
package main

import "fmt"

type stack []string

func (s *stack) Push(str string) {
	*s = append(*s, str)
}

func (s *stack) Pop() {
	*s = (*s)[:len(*s)-1]
}

func (s *stack) Peek() string {
	return (*s)[len(*s)-1]
}

func main() {
	var meinStack stack

	meinStack.Push("This")
	meinStack.Push("is")
	meinStack.Push("Sparta")
	fmt.Println(meinStack.Peek())
	meinStack.Pop()
	meinStack.Push("Brasil")

	fmt.Println(meinStack)
}

File: java/Stack.java
import java.util.ArrayList;
import java.util.List;

public class Stack {
  private static List<Integer> items = new ArrayList<>();

  public Stack() {}

  public void push(int item) {
    items.add(item);
  }

  public void pop() {
    items.remove(items.size() - 1);
  }

  public int peek() {
    return items.get(items.size() - 1);
  }

  public List<Integer> all() {
    return items;
  }
}
